### PR TITLE
Allow extrapolation of Astar as a function of muR in AnvredCorrections

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work.**
Allow extrapolation of cubic fits to A* as a function of muR - required for very absorbing samples sometimes measured on WISH. This is not a perfect solution (see issue #31342 for more details).

**To test:**
- All builds pass

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DannyHindson** [comment]
Just noticed your comment about "very absorbing samples" so seems like it is real

**Comment by RichardWaiteSTFC** [comment]
> Just noticed your comment about "very absorbing samples" so seems like it is real

Yeh it's not that often but they do measure samples with elements like manganese etc. also they do the correction on all bins so potentially up to quite large lambda >8 Ang!

File: Framework/Crystal/test/AnvredCorrectionTest.h [hunk 1]
```diff
@@ -138,6 +138,24 @@ class AnvredCorrectionTestPerformance : public CxxTest::TestSuite {
 
   void testNoEventsPerformace() { do_test_events(workspace, false, performance); }
 
+  void test_extrapolation_large_muR() {
```
Comment (DannyHindson) [inline Framework/Crystal/test/AnvredCorrectionTest.h:141]: Looks like this is currently in the performance test suite. I think it should be in the main AnvredCorrectionTest class?

**Review by DannyHindson — CHANGES_REQUESTED** [review]
Please can you move the unit test into the correct class? 

Also - possibly not related to your changes but I just noticed that the warning messages were only showing up for me in workbench if I set the log level to debug. Not sure why as you've clearly set them at warning level. Do you see this also? Will try again in the morning in case I've done something funny with my set up

File: Framework/Crystal/test/AnvredCorrectionTest.h (hunk 1 above)
Comment (RichardWaiteSTFC) [reply Framework/Crystal/test/AnvredCorrectionTest.h:141]: Indeed, don't know how I ended up putting it there sorry!

**Comment by RichardWaiteSTFC** [comment]
> Also - possibly not related to your changes but I just noticed that the warning messages were only showing up for me in workbench if I set the log level to debug. Not sure why as you've clearly set them at warning level. Do you see this also? Will try again in the morning in case I've done something funny with my set up

I can reproduce this on my branch - warnings from other algorithms appear to be OK (e.g. FindSXPeaks) - so I am not sure what is causing this.

**Review by DannyHindson — APPROVED** [review]
Tested with following script:
```
ws = CreateSampleWorkspace("Event",XMin=5000)
ws_lamba = ConvertUnits(ws, "Wavelength")
wsOut = AnvredCorrection(ws,LinearScatteringCoef=0.,
    LinearAbsorptionCoef=1.0,Radius=10.1,PowerLambda=3)
```
Log entries are output as required now

**Pull request merged**